### DISCUSSION TITLE
Failed to read build command file

### DISCUSSION THREAD

**Discussion by JaapWijnen** [SwiftPackageIndex/SwiftPackageIndex-Server#1880] [Q&A]
Hi! I'm getting a `error: failed to read build command file` error for my package which I've just submitted. I'm not sure what I'm doing wrong here. Any tips would be very welcome! the error can be found here: https://swiftpackageindex.com/builds/542006B8-3227-4407-B4A9-0F5377F2B263

**Comment by finestructure** [comment]
The error at the top just means that we're having trouble displaying the actual build command we're running. It is

```
env DEVELOPER_DIR="/Applications/Xcode-13.4.1.app" xcrun xcodebuild -IDEClonedSourcePackagesDirPathOverride="$PWD/.dependencies" -derivedDataPath "$PWD/.derivedData" build -scheme "XMLParser" -destination "generic/platform=ios"
```

which is also now showing up correctly after I've re-run the build: https://swiftpackageindex.com/builds/8F433C52-7BE3-426C-B2C6-60C29C2E3F5E.

The updated logs show there's a problem with scheme discovery:

```
xcodebuild: error: The workspace named "spi-builder-workspace" does not contain a scheme named "XMLParser". The "-list" option can be used to find the names of the schemes in the workspace.
```

The available schemes are:

```
Information about workspace "checkout":
    Schemes:
        _URLRouting
        CasePaths
        Collections
        CustomDump_watchOS
        DequeModule
        OrderedCollections
        Parsing
        Parsing_watchOS
        swift-case-paths-benchmark
        swift-case-paths-Package
        swift-collections-Package
        swift-custom-dump
        swift-parsing-benchmark
        swift-parsing-Package
        swift-xml-parser
        variadics-generator
```

**Reply by finestructure** [reply]
I think you want `swift-xml-parser` as the scheme in `.spi.yml`. I've opened a PR to make this change: https://github.com/JaapWijnen/swift-xml-parser/pull/2

**Reply by finestructure** [reply]
I'm not sure why the first run errored out like this. It's possible there's a problem with one of our runners - we'll need to keep on eye on this. (Unfortunately I deleted the build without grabbing a link to the logs 🤦‍♂️😅)

Thanks for reporting this!

**Reply by JaapWijnen** [reply]
Oh thanks for the PR! I copied the config from the pointfreeco/swift-parsing repo since (I think) I have a similar package setup. they use Parsing as their scheme name in their .spi.yml. Any idea what causes the different from my situation?
Thanks for the info!

**Reply by JaapWijnen** [reply]
Btw @finestructure is there a central place for the build system documentation? I'm trying to find what the platform keyword is to make the macOS Xcode build succeed.

**Reply by JaapWijnen** [reply]
Also I accepted the change! :) Let's see if it works once it picks up the rebuild

**Reply by finestructure** [reply]
> Btw @finestructure is there a central place for the build system documentation? I'm trying to find what the platform keyword is to make the macOS Xcode build succeed.

This here is currently still the best place: https://swiftpackageindex.com/docs/builds

We're tracking some improvements here: https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/issues/864

As to your question, the macOS platform keyword is `macos-xcodebuild`!

**Reply by finestructure** [reply]
> I copied the config from the pointfreeco/swift-parsing repo since (I think) I have a similar package setup. they use Parsing as their scheme name in their .spi.yml.

Yeah, I'd have done exactly the same. I don't understand why `xcodebuild -list` doesn't show your scheme as `XMLParser` the same way it does for `swift-parsing`. It's baffling - I can't see any differences in the package manifests that would explain it 🤔

**Reply by JaapWijnen** [reply]
> > I copied the config from the pointfreeco/swift-parsing repo since (I think) I have a similar package setup. they use Parsing as their scheme name in their .spi.yml.
> 
> Yeah, I'd have done exactly the same. I don't understand why `xcodebuild -list` doesn't show your scheme as `XMLParser` the same way it does for `swift-parsing`. It's baffling - I can't see any differences in the package manifests that would explain it 🤔

Hmm strange indeed!

**Reply by JaapWijnen** [reply]
> > Btw @finestructure is there a central place for the build system documentation? I'm trying to find what the platform keyword is to make the macOS Xcode build succeed.
> 
> This here is currently still the best place: https://swiftpackageindex.com/docs/builds
> 
> We're tracking some improvements here: #864
> 
> As to your question, the macOS platform keyword is `macos-xcodebuild`!

Awesome thanks, added that one as well! Ah ok, the FAQ also linked there indeed, but the url is /docs/builds so I thought maybe there's some central /docs page to gather all info relevant to package maintainers or something! But yeah .spi.yml documentation would be great! :) thanks for all the help!
